demisto_sdk/commands/common/files/file.py

```diff
@@ -0,0 +1,373 @@
+                clear_cache=clear_cache,
+            )
+        except FileReadError as e:
+            logger.warning(
+                f"Received error {e} when trying to retrieve {git_path_url} content from Github, retrying"
+            )
+            try:
+                return cls.read_from_http_request(
+                    git_path_url,
+                    params=frozenset({"token": github_token}.items()),
+                    timeout=timeout,
+                )
+            except FileReadError:
+                logger.exception(
+                    f"Could not retrieve the content of {git_path_url} file from Github"
+                )
+                raise
+
+    @classmethod
+    def read_from_gitlab_api(
+        cls,
+        path: str,
+        git_content_config: Optional[GitContentConfig] = None,
+        tag: str = DEMISTO_GIT_PRIMARY_BRANCH,
+        handler: Optional[XSOAR_Handler] = None,
+        clear_cache: bool = False,
+    ) -> Any:
+        if not git_content_config:
+            git_content_config = GitContentConfig()
+
+        git_path_url = urljoin(
+            git_content_config.base_api, "files", urllib.parse.quote_plus(path), "raw"
+        )
+        gitlab_token = git_content_config.CREDENTIALS.gitlab_token
+
+        return cls.read_from_http_request(
+            git_path_url,
+            headers=frozenset({"PRIVATE-TOKEN": gitlab_token}.items()),
+            params=frozenset({"ref": tag}.items()),
+            handler=handler,
+            clear_cache=clear_cache,
+        )
+
+    @classmethod
+    @lru_cache
+    def read_from_http_request(
+        cls,
+        url: str,
+        headers: Optional[frozenset] = None,
+        params: Optional[frozenset] = None,
+        verify: bool = True,
+        timeout: Optional[int] = None,
+        handler: Optional[XSOAR_Handler] = None,
+        clear_cache: bool = False,
+    ) -> Any:
+        if cls is File:
+            raise ValueError(
+                "when reading from file content please specify concrete class"
+            )
+        if clear_cache:
+            cls.read_from_http_request.cache_clear()
+        try:
+            response = requests.get(
+                url,
+                params={key: value for key, value in params} if params else None,
+                verify=verify,
+                timeout=timeout,
+                headers={key: value for key, value in headers} if headers else None,
+            )
+            response.raise_for_status()
+        except RequestException as e:
+            logger.exception(f"Could not retrieve file from {url}")
+            raise HttpFileReadError(url, exc=e)
+
+        try:
+            return cls.read_from_file_content(response.content, handler=handler)
+        except FileContentReadError as e:
+            logger.exception(f"Could not read file from {url} as {cls.__name__} file")
+            raise HttpFileReadError(url, exc=e)
+
+    @classmethod
+    def write_file(
```

Comment:
what is the difference between this and `write`?

Reply:
`write_file` already creates the instance for you, so you can just call for example `TextFile.write_file(...)` and it would write whatever you need in one-line.